### ISSUE TITLE
Couchbase Lite REST API calls partly implemented in PhoneGap plugin

### ISSUE THREAD

**Issue by Crapulax** [couchbaselabs/Couchbase-Lite-PhoneGap-Plugin#59]
Hello,

Using version 1.1.1

Unless I am doing something wrong, it seems that part of the REST couchbase lite API is not available using the couchbase lite PhoneGap plugin. For instance I am triying to use GET /_all_dbs but it fails with message:
return this.makeRequest('GET', this.databaseUrl + '/_all_dbs');
===> error 400: Invalid database

I was willing to use above to get the name of the replication DB. I tried previously to access _all_docs from  _replicator database but I received an error:
return this.makeRequest("GET", this.databaseUrl + '_replicator' + "/_all_docs");
===> "Router unable to route request to DO_GET_replicator"

I tried to access the /_config DB to change some settings but this one is not routed too. Note here that I found this in CouchDB documentation and might not be available at all.

My questions are:
1- Am I doing something wrong?
2- Do you plan to implement them? Alternatively may I get some information on where I can implement them?

The underlying of above requests is that we need to know the connection status with the couchbase sync gateway. We have currently implemented a periodic call to the sync with the GET {} to achieve it but this is sub optimal and generates unnecessary traffic.

Thanks!

**Comment by Crapulax** [comment]
_all_dbs do work in fact. I was putting an extra / (/_all_dbs)
/_config as explain here https://github.com/couchbase/couchbase-lite-ios/wiki/Guide%3A-Differences-From-CouchDB there is no implementation of /_config

So the good news is that it is working as expected. But still remain what I want to do. I need to access the replication state. I have found out that the couchbase lite db request changes to the sync gateway passing a document id on <db>/_local/<replication doc id>. How can I get this id from couchbase lite? (currently i get it from sync logs..)

**Comment by Crapulax** [comment]
tags get removed please read:
passing a document id on "mydb"/_local/"replication doc id". How can I get this id from couchbase lite? (currently i get it from sync logs..)

**Comment by Crapulax** [comment]
For the sake of clarity, above REST request was made against the sync gateway.

From couchbase lite i can get when setting up replication a repl001 doc id.
From couchbase lite I can get from root/_active_tasks statuses of replications.

So now I get my replication state. With one issue which is that the default heartbeat is set to 5mn (300000ms) and trying to change it by passing an additional params when setting up replication as {heartbeat : 60000} have no effect.

What I want to achieve is to know if there are pending messages to replicate. Was willing to do so by comparing available seq from the sync and last seq from couchbase lite. But I cannot get available seq from the sync without knowing the sync replication doc id. Information that couchbase lite knows as it pass it to setup the replication. But how can I get it from couchbase lite?
For whose who might ask, I need this to set an according badge value on app icons while running in background. And as far as I have seen, replication gets stopped while in background (which is the expected behavior).


Is there a smarter and feasible way to achieve this?

**Comment by jchris** [comment]
There is an _active_tasks api

**Comment by Crapulax** [comment]
Thanks for your answer. It does not solve my issue as _active_tasks does not provide the last sequence number replicated...
For others let me give you what I have understood (as documentation is humm let say incomplete).

1) As stated by jchris we can get a data flow of _active_tasks as we do get for _changes. This is useful to monitor the replication.
2) the replication identifiers used by couchbase lite is built using the dbuuid and replication data all this going to SHA1. Then the replication stores with id _local/<sha1 id> the replication id used to communicate with the replicated DB (the sync gateway in my case). _local/xxx document cannot be searched in any way. You can only get them if you know their id. So no way to get the replication id from their (unless you recompute the id, but well this is going to be painful and somehow risky).

In conclusion _active_tasks should return current sequences information instead of nearly unuseful data. By the way per couchDB documentation, _active_tasks return missing data.
